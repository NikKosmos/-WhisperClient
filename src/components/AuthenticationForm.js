import React from 'react'
import PropTypes from 'prop-types'

class AuthenticationForm extends React.Component {

    state = {
        email: '',
        password: '',
    }

    onSubmit = (event) => {
        event.preventDefault()

        this.props.onSubmit(this.state.email, this.state.password)

        this.setState({
            email: '',
            password: '',
        })
    }

    updateUserEmailValue = (eventArg) => {
        this.setState({
            email: eventArg.target.value
        })
    }

    updateUserPasswordValue = (eventArg) => {
        this.setState({
            password: eventArg.target.value
        })
    }

    render() {
        return (
            <form onSubmit = { this.onSubmit } className = "regAuthForm">
                <h3>Login</h3>
                <label>User email</label>
                <input
                    name = 'userEmail'
                    placeholder='User email'
                    type = 'text'
                    value = { this.state.email }
                    onChange = { this.updateUserEmailValue }
                /><br />
                <label>User password</label>
                <input
                    name = 'userPassword'
                    placeholder = 'User password'
                    type = 'text'
                    value = { this.state.password }
                    onChange = { this.updateUserPasswordValue }
                /><br />
                <button type = "submit">
                    Submit
                </button>
            </form>
        )
    }
}

AuthenticationForm.propTypes = {
    onSubmit: PropTypes.func.isRequired
}

export { AuthenticationForm }