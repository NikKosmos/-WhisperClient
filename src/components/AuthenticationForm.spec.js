import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json';

import { AuthenticationForm } from './AuthenticationForm'

describe('AuthenticationForm', () => {
    const mockOnSubmit = jest.fn()

    it ('should render correctly', () => {
        const testedComponent = shallow(<AuthenticationForm 
            onSubmit = { mockOnSubmit }
        />)

        expect(shallowToJson(testedComponent)).toMatchSnapshot();
      })

})

describe('when typing name', () => {
    const mockOnSubmit = jest.fn()
    const newTestUserEmail = 'capitan-nemo@google.com'

    const myComponent = shallow(<AuthenticationForm
        onSubmit = { mockOnSubmit }
    />)

    beforeEach(() => {
        myComponent.find('[name="userEmail"]').simulate('change', {
            target: {
                value : newTestUserEmail,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().email).toEqual(newTestUserEmail);
    })

})

describe('when typing email', () => {
    const mockOnSubmit = jest.fn()
    const newTestUserPassword = 'root'

    const myComponent = shallow(<AuthenticationForm
        onSubmit = { mockOnSubmit }
    />)

    beforeEach(() => {
        myComponent.find('[name="userPassword"]').simulate('change', {
            target: {
                value : newTestUserPassword,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().password).toEqual(newTestUserPassword);
    })

})

describe('when push the button', () => {
    const mockOnSubmit = jest.fn()

    const myComponent = shallow(<AuthenticationForm
        onSubmit = { mockOnSubmit }
    />)

    beforeEach(() => {
        myComponent.find('form').simulate('submit', {
            preventDefault: () => { },
        })
    })

    it('calls the props.onSubmit', () => {
        expect(mockOnSubmit).toHaveBeenCalledTimes(1);
    })

})