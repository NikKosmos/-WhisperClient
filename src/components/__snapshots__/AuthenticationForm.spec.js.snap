// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Login
  </h1>
  <label>
    User email
  </label>
  <input
    name="userEmail"
    onChange={[Function]}
    placeholder="User email"
    type="text"
    value=""
  />
  <br />
  <label>
    User password
  </label>
  <input
    name="userPassword"
    onChange={[Function]}
    placeholder="User password"
    type="text"
    value=""
  />
  <br />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
